### DISCUSSION TITLE
docs: Example config for Artifactory generic remote repository?

### DISCUSSION THREAD

**Discussion by mbx64** [briantist/galactory#139] [Q&A] [answered]
Could you please give an example on how to setup the generic remote repository on Artifactory? Which URL do I use and which layout?

**Comment by briantist** [comment]
Hi @mbx64 ,

There are no special requirements for the generic repository in Artifactory, just be aware that if the repository requires authentication to read from it, you will need to provide galactory with auth even just to read from it.

I don't really use layouts in Artifactory, so I would say use no layout if possible. It's also possible that if you do set a layout, it will be ignored.

The URL is direct URL to the repository, or to a subfolder within it.

If your artifactory base URL is `https://artifactory.example.com/artifactory` and your repository is called `repo1`, then you URL is likely `https://artifactory.example.com/artifactory/repo1`.

If repo1 contained other things and you wanted to use the subfolder `ansible` within that repository as the "base" for galactory, then your URL would be `https://artifactory.example.com/artifactory/repo1/ansible`

One thing to note: if you usually browse directly to `https://artifactory.example.com`, there's a chance that the URL still needs `/artifactory/` in it. I ran into an installation like this, where nginx was used to make URLs "work" without that path, but it caused strange issues with some API calls, so be aware. If you add `/artifactory/` to your "normal" URLs and they still work, you should probably use the longer version.

Hope this helps, would love to hear more about your use of galactory!

**Comment by mbx64** [comment]
Maybe I misunderstood this project. I was hoping to be able to download collections from https://galaxy.ansible.com/ through Artifactory as a proxy/mirror. But that's not what galactory is made for, right?

**Reply by briantist** [reply, marked as answer]
This what galactory is made for, but you cannot point `ansible-galaxy` directly at Artifactory to do so, you must run galactory as its own service. Galactory can proxy to an upstream galaxy server, and it can optionally cache the collections it proxies. "Caching" the collections in this case means storing them in Artifactory.

It is similar functionality to Artifactory's "virtual" repositories, but since Artifactory does not support Galaxy as a repository type, this project provides a Galaxy endpoint, while making use of your existing Artifactory infrastructure for storage.

Let me know if that clears it up and if you have any other questions.
